From 6b809649650c367ea0458928434600ca064206c1 Mon Sep 17 00:00:00 2001
From: Paul Barrette <paul.barrette@windriver.com>
Date: Fri, 26 Apr 2019 17:41:04 -0300
Subject: [PATCH] STX: qemu: add -enable-dpdk runtime flag

For Intel's dpdk openvswitch support, qemu takes a set of dpdk
specific command line args.  Qemu will dump and error if it doesn't
recognize the CLI args.  So, add a new runtime flag to tell qemu that
it is using dpdk and then the main function can take the appropriate
action.  This way, qemu can take dpdk args or run normally without
dpdk.

Signed-off-by: Paul Barrette <paul.barrette@windriver.com>
Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>
[ Update qemu-options.hx struc ]
Signed-off-by: Rafael Falcao <Rafael.VieiraFalcao@windriver.com>
---
 qemu-options.hx | 7 +++++++
 softmmu/vl.c    | 1 +
 2 files changed, 8 insertions(+)

diff --git a/qemu-options.hx b/qemu-options.hx
index 104632ea34..1a339453e4 100644
--- a/qemu-options.hx
+++ b/qemu-options.hx
@@ -906,6 +906,13 @@ DEFHEADING()
 
 DEFHEADING(Block device options:)
 
+DEF("enable-dpdk", 0, QEMU_OPTION_enable_dpdk,
+    "-enable-dpdk initialized DPDK EAL environment\n", QEMU_ARCH_ALL)
+SRST
+``-enable-dpdk``
+    Enable DPDK EAL environment at init time (expecting DPDK CLI params).
+ERST
+
 DEF("fda", HAS_ARG, QEMU_OPTION_fda,
     "-fda/-fdb file  use 'file' as floppy disk 0/1 image\n", QEMU_ARCH_ALL)
 DEF("fdb", HAS_ARG, QEMU_OPTION_fdb, "", QEMU_ARCH_ALL)
diff --git a/softmmu/vl.c b/softmmu/vl.c
index 5f556f1a0e..a398697d0d 100644
--- a/softmmu/vl.c
+++ b/softmmu/vl.c
@@ -3899,6 +3899,7 @@ void qemu_init(int argc, char **argv, char **envp)
                 qsp_enable();
                 break;
             case QEMU_OPTION_nouserconfig:
+            case QEMU_OPTION_enable_dpdk:
                 /* Nothing to be parsed here. Especially, do not error out below. */
                 break;
             default:
-- 
2.25.1
